websites/W/Warriorcats/presence.ts

```diff
@@ -0,0 +1,270 @@
+const presence = new Presence({
+		clientId: "936630095035113522",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+async function getPostNumber() {
+	// find <article></article> and get the #post- number
+	let postNumber = document.querySelector("article").getAttribute("id");
+	postNumber = postNumber.replace(/post-/, "");
+	return postNumber;
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		details: "Viewing an unsupported page",
+		state: "",
+		largeImageKey:
+			"https://upload.wikimedia.org/wikipedia/commons/thumb/c/cf/Warrior_Cats_logo.svg/2560px-Warrior_Cats_logo.svg.png",
+		startTimestamp: browsingTimestamp,
+		buttons: [
+			{ label: "Current Page", url: "https://www.warriorcats.com/" },
+			{ label: "Warrior Cats Germany", url: "https://www.warriorcats.de/" },
+		],
+	};
+
+	switch (document.location.pathname) {
+		case "/": {
+			presenceData.details = "Viewing";
+			presenceData.state = "Warrior Cats Home";
+			presenceData.largeImageKey =
+				"https://www.warriorcats.de/wp-content/uploads/9783407759085-web.jpg";
+			presenceData.buttons = [
+				{ label: "Current Page", url: document.location.href },
```

Comment:
Rather than a generic "View Current Page" button, prefer using buttons with specific labels and which lead to useful content that a user may want to visit.